### PR TITLE
Initial Fuzzing Integration & OSS-Fuzz scripts

### PR DESCRIPTION
**Pull request by DaveLak**
@davvid & @Theelx, this introduces fuzzing test harness and supporting files OSS-Fuzz integration scripts  as discussed in issue https://github.com/jsonpickle/jsonpickle/issues/496. The changes proposed in this PR are documented in the `fuzzing/README.md` so that's probably the best place to start if you haven't already seen it.

I've also put up a draft for the corresponding OSS-Fuzz PR here: https://github.com/google/oss-fuzz/pull/12571.

Please note that CI will continue to fail on that PR until this one is merged because it depends on the scripts added here.

Additionally I still need to add @Theelx to the auto_cc list in the `project.yaml` file which enables access to the issue tracker. Would you let me know which Gmail address you'd like added?

Let me know what you think!

### CODE REVIEW

**Comment by Theelx** [comment]
@DaveLak Thank you so much for this PR! I'll take a look shortly. You can add ``theelgirl4@gmail.com`` to the auto_cc list. I'll need to set up an automatic forwarding rule from that to my main email though, and a way to filter out spammers now that it's public. Do you have any suggestions for spam filter rules?

**Comment by DaveLak** [comment]
Thanks, @Theelx! I'll get it added.

> Do you have any suggestions for spam filter rules?

Honestly no, not really 😅 For whatever reason, I haven't had many issues with spam on any of the email accounts I have on GitHub. Hopefully you won't either!

File: fuzzing/oss-fuzz-scripts/container-environment-bootstrap.sh
Comment (Theelx) [inline fuzzing/oss-fuzz-scripts/container-environment-bootstrap.sh:78]: Should atheris be 'atheris~=2.3', or is there a reason why you did >=2.3.0?

File: .gitignore
Comment (Theelx) [inline .gitignore:31]: Do we also want to ignore the .dict files that seem to be produced in the bash scripts?

**Review by Theelx — APPROVED** [review]
A few minor nitpicks and questions, but looks great overall!

File: fuzzing/oss-fuzz-scripts/container-environment-bootstrap.sh
Comment (DaveLak) [reply fuzzing/oss-fuzz-scripts/container-environment-bootstrap.sh:72]: > In one of your commits, you removed the function to create a seed corpora with the following description

Good eyes, thanks for giving this such a careful review!

> Can you either explain to us how to set up the seed corpora, or put back the function that lets us create such a corpora, even if it isn't used quite yet?

Absolutely! In fact I planned to offer both options up for consideration, but the corpora is more optimization than requirement so I worried introducing another decision point may distract from the more important goal of getting the fuzzer fuzzing.

But you're right to ask about. I should have at least mentioned that commit in the PR description.

I'll included more details in a follow-up comment shortly!

File: fuzzing/README.md
Comment (DaveLak) [reply fuzzing/README.md:46]: Yes, it's very possible and can actually be extremely easy. Unfortunately, the inverse is also true though. It mostly depends on the the computer and Clang/llvm version available.

I didn't include the direct execution steps because there are a lot of variables. The README is mostly written for a general Python audience with little to no fuzzing experience; I was reduce the potential friction points as much as reasonably possible to encourage new contributors to feel comfortable improving/fixing the tests.

But you can give it a shot right now and let me know if I the README should be updated 🙂 :

```
## Running Fuzzers Locally

### Direct Execution of Fuzz Targets

For quick testing of changes, [Atheris](https://github.com/google/atheris) makes it possible to execute a fuzz target directly:

1. Install Atheris following the [installation guide](https://github.com/google/atheris) for your operating system.
2. Execute a fuzz target, for example:

```shell
python fuzzing/fuzz-targets/fuzz_config.py
```
```

Do note though: the OSS-Fuzz setup does a lot of other things beyond simple input mutation and it's designed to run distributed at massive scale, so Docker is a requirement for running it & debugging it.

**Review by davvid — APPROVED** [review]
Thanks again! Regarding docker (and sorry for the naive question) should we mostly expect the same setup to run under `podman` with minimal effort? I guess I can answer that myself once I have some time to give it a whirl, but just wondering if you happen to know.

Thanks for reviewing @Theelx -- I concur 100% w/ your notes. This is looking good enough to merge but we'll wait a bit to give you a chance to address some of the sugs when you've got some time. Thanks again for the fuzzing setup!

**Comment by davvid** [comment]
@DaveLak I know you're busy so thanks again, and please feel free to submit a follow-up and keep the conversation going.

Until then, I've merged this and applied the following commits per @Theelx 's review:
* fb9e1a0473c08a9e784153af9b91a9d600871ef2
* ee19e7676555c56a43363e3aa1f23178742b3310
* db3abd98d8e85d089b6b15604fa8f0e1122e3159
* e53797324a8f6a4d2156c31a6f87f3f2a3f8faa1

**Comment by DaveLak** [comment]
Thanks, @davvid! Apologies for the delay.

Now that this is merged, https://github.com/google/oss-fuzz/pull/12571 can move forward. I've updated the config there to include @Theelx’s email for tracker notifications and marked it ready for OSS-Fuzz maintainer review.

Once merged, it may take a few days to start running. While I don’t have full visibility due to not being listed in the config, I’ll provide links and guidance to help you both get up to speed.

I’ve also drafted some documentation addressing @Theelx’s questions about dictionaries and seed corpora, which I’ll share soon.

**Pull request merged**
